service/controller/app.go

```diff
@@ -0,0 +1,122 @@
+package controller
+
+import (
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/app-operator/service/controller/v1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/informer"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/kubernetes"
+	//"github.com/giantswarm/app-operator/service/controller/v1"
+)
+
+const AppControllerSuffix = "-app"
+
+type AppConfig struct {
+	G8sClient    versioned.Interface
+	K8sClient    kubernetes.Interface
+	K8sExtClient apiextensionsclient.Interface
+	Logger       micrologger.Logger
+
+	ProjectName    string
+	WatchNamespace string
+}
+
+type App struct {
+	*controller.Controller
+}
+
+func NewApp(config AppConfig) (*App, error) {
+	var err error
+
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.G8sClient must not be empty")
```

Comment:
```suggestion
return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
```

This is outdated. We should use the new approach. It makes it easier to read stack traces when debugging.